File: frontend/app/lib/aiLanguage.ts
export type AiLanguage = "en" | "hi" | "or";

export const AI_LANGUAGES: Array<{ value: AiLanguage; label: string }> = [
  { value: "en", label: "EN" },
  { value: "hi", label: "हिंदी" },
  { value: "or", label: "ଓଡ଼ିଆ" },
];

export const AI_LANGUAGE_STORAGE_KEY = "nirikshan.ai-language";

File: src/main/java/com/nirikshan/model/AiLanguage.java
package com.nirikshan.model;

import java.util.Locale;

public enum AiLanguage {
    EN("en", "English"),
    HI("hi", "Hindi"),
    OR("or", "Odia");

    private final String code;
    private final String displayName;

    AiLanguage(String code, String displayName) {
        this.code = code;
        this.displayName = displayName;
    }

    public String code() { return code; }
    public String displayName() { return displayName; }

    public static AiLanguage fromCode(String value) {
        if (value == null || value.isBlank()) return EN;
        String normalized = value.trim().toLowerCase(Locale.ROOT);
        for (AiLanguage language : values()) {
            if (language.code.equals(normalized)) return language;
        }
        throw new IllegalArgumentException("language must be one of: en, hi, or");
    }
}
